src/keymanager.h

```diff
@@ -4,20 +4,30 @@
 #include <boost/move/unique_ptr.hpp>
 #include <boost/shared_ptr.hpp>
 
-#include "config.h"
 #include "httpinterface.h"
 #include "invstorage.h"
-#ifdef BUILD_P11
 #include "p11engine.h"
-#endif
+#include "types.h"
 #include "utils.h"
 
+// bundle some parts of the main config together
+struct KeyManagerConfig {
+  KeyManagerConfig() = delete;
```

Comment:
If thre is no any other constructors how it is possible to instantiate this struct?

Reply:
It should be derived from one of the main configs: `Config` and `AktualizrSecondaryConfig`. I didn't want it to have default value. So the way to construct it is `KeyManagerConfig kmconfig{field_values,...}`.

But yes, deleting the default constructor is not strictly needed and can be confusing. Maybe I'll replace it with a comment.